api/src/main/java/org/openmrs/util/ConceptReferenceRangeUtility.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.util;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+import org.apache.velocity.exception.ParseErrorException;
+import org.joda.time.LocalTime;
+import org.openmrs.Concept;
+import org.openmrs.Obs;
+import org.openmrs.Person;
+import org.openmrs.api.APIException;
+import org.openmrs.api.context.Context;
+
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A utility class that evaluates the concept ranges 
+ * 
+ * @since 2.7.0
+ */
+public class ConceptReferenceRangeUtility {
+	
+	public ConceptReferenceRangeUtility() {
+	}
+	
+	/**
+	 * This method evaluates the given criteria against the provided person/patient.
+	 *
+	 * @param criteria the criteria string to evaluate e.g. "$patient.getAge() > 1"
+	 * @param obs The observation (Obs) object containing the values to be used in the criteria.
+	 *                  
+	 * @return true if the criteria evaluates to true, false otherwise
+	 */
+	public boolean evaluateCriteria(String criteria, Obs obs) {
+		if (obs == null) {
```

Comment:
Shouldn't we be able to pass null for obs? For cases that do not need it like evaluations that deal with only the person object like age?

Reply:
In what circumstance do we expect to have person object without `Obs`? 

The main aim for this check was to avoid null pointer in `obs.getPerson()`.